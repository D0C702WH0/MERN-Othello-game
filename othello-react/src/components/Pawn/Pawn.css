
.pawn {
  border-radius: 50%;
  margin: 0 auto;
}

.black {
  border: 1px solid white;
  box-shadow: 3px 3px 3px black;
  background-color: black;
}

.white {
  border: 1px solid black;
  box-shadow: 3px 3px 3px black;
  background-color: white;
}

@media only screen and (min-width:768px) {
  .pawn {
    min-width: 3rem;
    min-height: 3rem;
  }
}

@media only screen and (min-width:0px) and (max-width:767px) {
  .pawn {
    min-width: 1.8rem;
    min-height: 1.8rem;
  }
}