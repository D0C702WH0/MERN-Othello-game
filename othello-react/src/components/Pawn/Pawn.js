import React from 'react'
import PropTypes from 'prop-types'

import './Pawn.css'

export default function Pawn ({ color }) {
  return (
    <div className={`pawn ${color}`} />
  )
}

Pawn.propTypes = {
  color: PropTypes.string
}
