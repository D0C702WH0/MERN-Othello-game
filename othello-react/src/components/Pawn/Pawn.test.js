import React from 'react'
import TestRenderer from 'react-test-renderer'
import Pawn from './Pawn'

let renderer
beforeEach(() => {
  renderer = TestRenderer.create(
    <Pawn color={'black'}/>
  )
})

describe('Pawn', () => {
  it('Basic rendering', () => {
    const result = renderer.toJSON()
    expect(result).toMatchSnapshot()
  })

  it('Should have "black" as a className', () => {
    const instance = renderer.root
    const element = instance.findByType('div')
    expect(element.props.className.includes('black')).toBe(true)
  })

  it('Should change "black" className to white on props change', () => {
    renderer.update(<Pawn color={'white'}/>)
    const instance = renderer.root
    const element = instance.findByType('div')
    const result = renderer.toJSON()
    expect(result).toMatchSnapshot()
    expect(element.props.className.includes('white')).toBe(true)
  })
})
