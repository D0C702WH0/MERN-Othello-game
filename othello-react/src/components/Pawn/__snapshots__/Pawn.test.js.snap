// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pawn Basic rendering 1`] = `
<div
  className="pawn black"
/>
`;

exports[`Pawn Should change "black" className to white on props change 1`] = `
<div
  className="pawn white"
/>
`;
